packages/material-ui/src/useMediaQuery/useMediaQuery.js

```diff
@@ -1,25 +1,12 @@
 import * as React from 'react';
-import { useTheme } from '@material-ui/private-theming';
-import { unstable_useThemeProps as useThemeProps } from '@material-ui/system';
+import { unstable_useThemeProps as useThemeProps, useTheme } from '@material-ui/system';
 import useEnhancedEffect from '../utils/useEnhancedEffect';
 
 export default function useMediaQuery(queryInput, options = {}) {
   const theme = useTheme();
   // eslint-disable-next-line material-ui/mui-name-matches-component-name
   const props = useThemeProps({ name: 'MuiUseMediaQuery', props: {} });
 
-  if (process.env.NODE_ENV !== 'production') {
-    if (typeof queryInput === 'function' && theme === null) {
-      console.error(
-        [
-          'Material-UI: The `query` argument provided is invalid.',
-          'You are providing a function without a theme in the context.',
-          'One of the parent elements needs to use a ThemeProvider.',
-        ].join('\n'),
-      );
-    }
-  }
-
```

Comment:
This change is missing a description

Reply:
Right, my bad. The `theme` can no longer be `null`. The `useTheme` from the system provides a default value for the theme if no theme is available in the context.

Reply:
I reverted this change. I've added a new `useThemeWithoutDefault` hook in the `@material-ui/system`, that wouldn't add the default theme in it. Also, I've replaced the `useThemeProps` with `getThemeProps`. Here is how it looks now:

![image](https://user-images.githubusercontent.com/4512430/122435040-ef7ead80-cf97-11eb-8f1a-cf1a05fe257a.png)

Let me know what you think

Reply:
I should have taken a screenshot when I left my message in https://github.com/mui-org/material-ui/pull/26793#discussion_r653497618. I saw the exact same negative diff as in your screenshot.

Reply:
if `useMediaQuery` depends only on `@material-ui/private-theming` than it means that it doesn't take into account the emotion/styled-components context, hence the smaller bundle size

Reply:
The `@material-ui/system`'s `useTheme` combines these two contexts. The `ThemeProvider` anyway combines the two contexts, so maybe the `useTheme` can depend only on the `private-theming` context. Let me try that.

Reply:
It's definitelly better:

![image](https://user-images.githubusercontent.com/4512430/122441340-c06b3a80-cf9d-11eb-8795-fcd2e84e4c2f.png)
 I am pushing last changes for fixing the `ThemeProvider` usages and I think we are good to go :)